config/pmd-test.xml

```diff
@@ -86,19 +86,10 @@
             <!-- in AstRegressionTest pmd does not find asserts in inner classes methods called from the test method -->
             <!-- in AllChecksTest pmd does not find asserts in lambdas -->
             <!-- in DetailASTTest pmd does not find asserts in lambdas -->
-            <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='SuppressionFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCode']
-            | //ClassOrInterfaceDeclaration[@Image='SuppressionCommentFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCodeOfTagClass']
-            | //ClassOrInterfaceDeclaration[@Image='SuppressWithNearbyCommentFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCodeOfTagClass']
-            | //ClassOrInterfaceDeclaration[@Image='SuppressionXpathFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCode']
-            | //ClassOrInterfaceDeclaration[@Image='XpathFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCode']
```

Comment:
The expression changed to "any method with a call to method `verify(...)` or `exit.checkAssertionAfterwards(...)` or `.verify(...)`.

`PackageObjectFactoryTest` and `XdocsPagesTest` do not use any of them, so they are explicit here.

Reply:
If we can verify that a test method calls a specific method, should we give these nested methods a special name to show they do assertions? Something like like `{testName}Assert(...)`.
I'm not sure if suppression model can do a regexp match.

Reply:
Done. Now it looks very funny. The rule named `JUnitTestsShouldIncludeAssert` is suppressed for any method with a call to `assert`.

Reply:
@pbludov This was more a future item, hence why I said give these secondary methods the name `*Assert`. If you want to undo it and make an issue, it is ok.

Reply:
Done. Now all the offenders are listed explicitly.